lib/matplotlib/testing/compare.py

```diff
@@ -367,12 +367,8 @@ def calculate_rms(expectedImage, actualImage):
         raise ImageComparisonFailure(
             "Image sizes do not match expected size: {0} "
             "actual size {1}".format(expectedImage.shape, actualImage.shape))
-    num_values = expectedImage.size
-    abs_diff_image = abs(expectedImage - actualImage)
-    histogram = np.bincount(abs_diff_image.ravel(), minlength=256)
-    sum_of_squares = np.sum(histogram * np.arange(len(histogram)) ** 2)
-    rms = np.sqrt(float(sum_of_squares) / num_values)
-    return rms
+    # Convert to float to avoid overflowing finite integer types.
+    return np.sqrt(((expectedImage - actualImage).astype(float) ** 2).mean())
```

Comment:
This is seems to be a substantive change to a genuine RMS, from something I don't understand.  There must have been a reason for the convoluted original.

Reply:
I don't understand the original either, but the relevant tests are still passing...

Reply:
Ours are, but I suspect this might cause problems for packagers who put in larger tolerances than ours, as I believe some do.  Therefore I recommend sticking to the original algorithm for the purposes of this PR.  Perhaps add a comment that this is *not* a simple rms.  @mdboom can probably tell us the rationale for the algorithm.

Reply:
AFAICT the results should be the same: the old method counted, for each delta, the number of pixels that have this delta value, then computed a RMS based on that histogram.
Again, https://github.com/matplotlib/matplotlib/pull/10335/files shows tests where we check that the reported RMS values are stable to 1e-4.

Reply:
If I am correctly expressing the two algorithms below, then their results are not the same:
```
In [26]: x = np.zeros(1000, dtype=int); x[0] = 1

In [27]: h = np.bincount(x, minlength=256)

In [28]: np.sqrt(((h * np.arange(len(h)))**2).mean())
0.0625

In [29]: np.sqrt((x**2).mean())
0.031622776601683791

In [30]: x[:10]
array([1, 0, 0, 0, 0, 0, 0, 0, 0, 0])

In [31]: (h * np.arange(len(h)))[:10]
array([0, 1, 0, 0, 0, 0, 0, 0, 0, 0])

In [32]: len(x)
1000

In [33]: len(h)
256
```

Reply:
the previous algorithm was dividing by num_values (so sum() / len(x) in your case, not mean()).  this fix shows that the values are indentical.

Reply:
Oops... sorry I missed that.

Reply:
Well, I can run the Cartopy test suite without any issue using this PR.